src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/RazorHtmlCSharpLanguageServer.cs

```diff
@@ -81,6 +81,31 @@ public Task ExitAsync(CancellationToken cancellationToken)
             return Task.CompletedTask;
         }
 
+        [JsonRpcMethod(Methods.TextDocumentDidOpenName, UseSingleObjectParameterDeserialization = true)]
```

Comment:
Why do we need all of these?

Reply:
According to Christian "textDocument/open" and "textDocument/close" need to be implemented for the flow on their end. I've asked him if that's still the case.

Reply:
Our Razor language server has these so you should  be able to remove these as well.